pandas/core/arrays/categorical.py

```diff
@@ -387,6 +387,14 @@ def dtype(self) -> CategoricalDtype:
     def _constructor(self) -> Type["Categorical"]:
         return Categorical
 
+    @classmethod
+    def _from_scalars(cls, data, dtype):
+        # if not all(
+        #     isinstance(v, dtype.categories.dtype.type) or isna(v) for v in data
+        # ):
+        #     raise TypeError("Requires dtype scalars")
```

Comment:
`if not all (x in dtype.categories or is_valid_nat_for_dtype(x, dtype.categories.dtype) for x in data)`?

Reply:
Yes, as mentioned somewhere in https://github.com/pandas-dev/pandas/pull/38315#issuecomment-739255818, for categorical we probably want to check if the values are valid categories.

We might already have some functionality for this? Like the main constructor, but then raising an error instead of coercing unknown values to NaN:

```
In [17]: pd.Categorical(["a", "b", "c"], categories=["a", "b"])
Out[17]: 
['a', 'b', NaN]
Categories (2, object): ['a', 'b']
```

The above is basically done by ``_get_codes_for_values``, so we might want a version of that which is strict instead of coercing to NaN.

Reply:
I think Categorical._validate_setitem_value doest what you're describing